internal/core/config.go

```diff
@@ -19,10 +19,29 @@
 package core
 
 import (
+	"regexp"
+
 	"github.com/sapcc/castellum/internal/db"
 )
 
 // Config contains the app-level configuration options.
 type Config struct {
-	MaxAssetSize map[db.AssetType]*uint64
+	MaxAssetSize []MaxAssetSizeRule
+}
+
+type MaxAssetSizeRule struct {
+	Regex *regexp.Regexp
+	Value *uint64
+}
+
+func (c Config) MaxAssetSizeFor(AssetType db.AssetType) (maxAssetSize *uint64) {
+	for _, MaxAssetSize := range c.MaxAssetSize {
```

Comment:
Names starting in upper case are usually reserved for public types/variables/functions. Since this variable is of type `MaxAssetSizeRule`, I'd suggest calling it `rule`.